sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -94,6 +94,62 @@ case class UserDefinedGenerator(
 }
 
 /**
+ * Separate v1, ..., vk into n rows. Each row will have k/n columns. n must be constant.
+ * {{{
+ *   SELECT stack(2, 1, 2, 3)) ->
+ *   1      2
+ *   3      NULL
+ * }}}
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(n, v1, ..., vk) - Separate v1, ..., vk into n rows.",
+  extended = "> SELECT _FUNC_(2, 1, 2, 3);\n  [1,2]\n  [3,null]")
+case class Stack(children: Seq[Expression])
+    extends Expression with Generator with ImplicitCastInputTypes with CodegenFallback {
+
+  override def inputTypes: Seq[DataType] =
+    Seq(IntegerType) ++ Seq.fill(children.length - 1)(children.tail.head.dataType)
```

Comment:
Can you check what's the type coercion rule for hive? It looks to me that the values for same column should be same type, but not all values should be same type.